vizro-core/tests/unit/vizro/actions/test_actions_utils.py

```diff
@@ -3,38 +3,60 @@
 from vizro.actions._actions_utils import _create_target_arg_mapping, _update_nested_graph_properties
 
 
-@pytest.fixture
-def fake_graph_property_dict():
-    return {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}}}
-
-
 class TestUpdateNestedGraphProperties:
     def test_update_nested_graph_properties_single_level(self):
         graph = {"color": "blue"}
         result = _update_nested_graph_properties(graph, "color", "red")
         expected = {"color": "red"}
         assert result == expected
 
-    def test_update_nested_graph_properties_multiple_levels(self):
-        graph = {"node": {"label": "A", "color": "blue"}}
-        result = _update_nested_graph_properties(graph, "node.color", 2)
-        expected = {"node": {"label": "A", "color": 2}}
+    @pytest.mark.parametrize(
+        "graph, dot_separated_strings, value, expected",
+        [
+            ({"node": {"label": "A", "color": "blue"}}, "node.color", "red", {"node": {"label": "A", "color": "red"}}),
+            (
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}}},
+                "nodes.A.color",
+                "red",
+                {"nodes": {"A": {"color": "red"}, "B": {"color": "green"}}},
+            ),
+        ],
+    )
+    def test_update_nested_graph_properties_multiple_levels(self, graph, dot_separated_strings, value, expected):
+        result = _update_nested_graph_properties(graph, dot_separated_strings, value)
         assert result == expected
 
-    def test_update_nested_graph_properties_nested_dict(self):
-        graph = {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}}}
-        result = _update_nested_graph_properties(graph, "nodes.A.color", "red")
-        expected = {"nodes": {"A": {"color": "red"}, "B": {"color": "green"}}}
+    @pytest.mark.parametrize(
+        "graph, dot_separated_strings, value, expected",
+        [
+            (
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}}},
+                "nodes.C.color",
+                "red",
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}, "C": {"color": "red"}}},
+            ),
+            (
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "red"}}},
+                "nodes.B.value",
+                "red",
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "red", "value": "red"}}},
+            ),
+            (
+                {"nodes": {"A": {"color": "blue"}, "B": {"color": "green"}}},
+                "nodes.B",
+                "red",
+                {"nodes": {"A": {"color": "blue"}, "B": "red"}},
+            ),
+        ],
+    )
+    def test_update_nested_graph_properties_add_keys(self, graph, dot_separated_strings, value, expected):
```

Comment:
```suggestion
    def test_update_nested_graph_properties_add_or_overwrite_keys(self, graph, dot_separated_strings, value, expected):
```

Reply:
And could we also add an empty `graph`, that I think is the last remaining case that might happen right?